From 1c3af36c14cae2d94b9ca0a2e4f6ad0fd38f6241 Mon Sep 17 00:00:00 2001
From: Akash Yadav <contact@itsaky.com>
Date: Fri, 13 Mar 2026 19:15:23 +0530
Subject: [PATCH 5/7] fix: avoid usage of javax.swing

Signed-off-by: Akash Yadav <contact@itsaky.com>
---
 .../impl/base/permissions/KaBaseAnalysisPermissionChecker.kt  | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/analysis/analysis-api-impl-base/src/org/jetbrains/kotlin/analysis/api/impl/base/permissions/KaBaseAnalysisPermissionChecker.kt b/analysis/analysis-api-impl-base/src/org/jetbrains/kotlin/analysis/api/impl/base/permissions/KaBaseAnalysisPermissionChecker.kt
index 17fe373e1130..670ed24ebb02 100644
--- a/analysis/analysis-api-impl-base/src/org/jetbrains/kotlin/analysis/api/impl/base/permissions/KaBaseAnalysisPermissionChecker.kt
+++ b/analysis/analysis-api-impl-base/src/org/jetbrains/kotlin/analysis/api/impl/base/permissions/KaBaseAnalysisPermissionChecker.kt
@@ -53,8 +53,8 @@ internal class KaBaseAnalysisPermissionChecker : KaAnalysisPermissionChecker {
     }
 
     private fun isProhibitedEdtAnalysis(application: Application): Boolean =
-        application.isDispatchThread &&
-                !permissionOptions.defaultIsAnalysisAllowedOnEdt &&
+        /*false &&
+                */!permissionOptions.defaultIsAnalysisAllowedOnEdt &&
                 !permissionRegistry.isAnalysisAllowedOnEdt
 
     private fun isProhibitedWriteActionAnalysis(application: Application): Boolean =
-- 
2.53.0

